MASFoundation/Classes/models/Network/MASRequest.h

```diff
@@ -0,0 +1,147 @@
+//
+//  MASRequest.h
+//  MASFoundation
+//
+//  Copyright (c) 2017 CA. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "MASClaims.h"
+#import "MASRequestBuilder.h"
+
+/**
+ MASRequest class is an object created by MASRequestBuilder. It's contains all parameters necessary to invoke an API.
+ The class cannot be constructed or changed directly, only through MASRequestBuilder.
+ */
+@interface MASRequest : NSObject
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+///--------------------------------------
+/// @name Properties
+///--------------------------------------
+
+# pragma mark - Properties
+
+/**
+ NSString value of the HTTP Method (GET, POST, PUT, DELETE).
+ */
+@property (nonatomic, strong, readonly) NSString *httpMethod;
+
+/**
+ BOOL value that determines whether or not the target host is a primary gateway or another gateway/public server.
```

Comment:
BOOL value that determines whether or not to include credentials of primary gateway in the request.